debitpolicy/debitpolicy.go

```diff
@@ -0,0 +1,115 @@
+package debitpolicy
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dogmatiq/dogma"
+	"github.com/dogmatiq/example/messages/commands"
+	"github.com/dogmatiq/example/messages/events"
+)
+
+// debitpolicy is the aggregate root for the debit policy.
+type debitpolicy struct {
+	// DebitAmount is the current sum of debit amounts for the period, in cents.
+	DebitAmount int64
+}
+
+func (p *debitpolicy) ApplyEvent(m dogma.Message) {
+	switch x := m.(type) {
+	case events.AccountDebitedForWithdrawal:
+		p.DebitAmount += x.Amount
+	case events.AccountDebitedForTransfer:
+		p.DebitAmount += x.Amount
+	}
+}
+
+// Aggregate implements the business logic for a debit policy.
+//
+// It centralizes all debits that are applied to an account over a period of
+// time in order to enforce a limited debit policy.
+type Aggregate struct{}
```

Comment:
I think this might be too generic, this isn't a policy for an arbitrary period of time, but rather it's specifically the `DailyDebitLimitPolicy`, or something like that.

Reply:
Good point. I wasn't sure what period to use so i left it open to interpretation for now. Daily sounds good, so i will rename that.